site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx

```diff
@@ -229,7 +229,16 @@ const CreateConsequenceServices = (props: CreateConsequenceServicesProps): React
     useEffect(() => {
         if (selectedService) {
             fetchStops(selectedService.id, pageState.inputs.vehicleMode, dataSource)
-                .then((stops) => setStopOptions(sortStops([...stopOptions, ...stops])))
+                .then((stops) =>
+                    setStopOptions(
+                        sortStops([
+                            ...stopOptions.filter(
```

Comment:
Can we move this sortStops and filter logic to fetchStops?

On line 651 we use fetchStops without sorting and filtering which means unordered and duplicate stops will appear if you go back to this page after entering data to edit your changes.

Reply:
Also, it might be a good idea to create a new function in the FormUtils file to do both the sorting and remove duplicate actions, following a similar patter to the filterServices function. That way we have a reusable piece of code that can easily be tested. 

I would use this function on this consequence page, and also use this new function in the ServicesMap file and the StopsMap file wherever we currently sort and filter stops.

Reply:
done